package/gluon-mesh-vpn-wireguard/files/usr/bin/wg-broker

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+run_broker() {
+  local interface="$1"
+  local pubkey="$2"
+  local port
+  local interval=5
+
+  group_peers=$(uci show wireguard|grep mesh_vpn_backbone_peer|grep =peer|cut -d"." -f2|cut -d= -f1)
+
+  for i in $group_peers
+  do
+    if [ $interface == $(uci get wireguard.$i.ifname) ]; then
+      thispeer=$i
+      break;
+    fi
+  done
+
+  localtime=$(date +%s)
+  if [[ -n $thispeer ]]; then
+
+    remote=$(uci get wireguard.$thispeer.remote)
+    brokerport=$(uci get wireguard.$thispeer.broker_port)
+    localport=$(wg show $interface listen-port)
+
+    serverreply="$(nc_timeout 10 $remote $brokerport "{\"pubkey\":\"$pubkey\", \"localport\":$localport }")"
+
+    if [[ "x$serverreply" != "x" ]]; then
+      port=$(jsonfilter -s "$serverreply" -e "@.port")
+      servertime=$(jsonfilter -s "$serverreply" -e "@.time")
+      if [[ "x$servertime" != "x" && $((servertime - localtime)) -gt 240 ]]; then
+	# local clock differs a lot from the server clock.
+	# assuming ntp is working only when a tunnel is established we need to
+	# set the clock to  something in the proximity of the correct time.
+	# Let's assume servertime for now. ntpd will handle the rest
+	formatted_time=$(date -d @$servertime +%Y%m%d%H%M.%S)
+	date -s $formatted_time >/dev/null
+      fi
+
+      if [[ -z $port ]]; then
+	error=$(jsonfilter -s "$serverreply" -e "@.error")
+	if [[ -n $error ]]; then
+	  reason=$(jsonfilter -s "$serverreply" -e "@.error.reason")
+	  ecode=$(jsonfilter -s "$serverreply" -e "@.error.code")
+	  echo "received error [$ecode] from host $remote: $reason" >&2
```

Comment:
Please use logger for printing the error messages.

Reply:
I do not agree with that one. logger is used exclusively in the tunneldigger and hoodselector implementations. The use case is to write to messages to the system log.

while the brokers logs mostly will land in system log this is not always the case, for example when calling the wireguard check scripts interactively. I'd rather keep it the way it is, writing error messages to stderr. If some framework is used therafter to write to syslog (like when calling wg-broker from netifd) then this is fine as well.

Reply:
`logger` can also log to stderr additionally with the '-s, --stderr' option. You get the log message in syslog and on stderr then.

Reply:
in that case, why not use echo bla >2 in the first place? What is the benefit for this use case?